### PR TITLE
[python] Add checker pd attr param and fix switch example

### LINKED ISSUE #7803: [Bug] [python] Missing set global parameter in switch task example
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

In switch example, we do not set global para for it, and it would fail while trying to run this example https://github.com/apache/dolphinscheduler/blob/946a0c7c5768506e7ca92a21e7aed6ad5aa60871/dolphinscheduler-python/pydolphinscheduler/examples/task_switch_example.py#L37-L40. Maybe we should add assert to check if param setted in https://github.com/apache/dolphinscheduler/blob/946a0c7c5768506e7ca92a21e7aed6ad5aa60871/dolphinscheduler-python/pydolphinscheduler/examples/task_switch_example.py#L44-L47

### What you expected to happen

Should hint user when global param not setted

### How to reproduce

run `python ./example/task_switch_example.py`

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhongjiajie**
* Correct submit self.param to java gateway
* Fix missing parameter for switch example
* Add mechanism checker before submit to java gateway

close: #7803

### CODE REVIEW

**Comment by zhongjiajie** [comment]
@devosend  PTAL if you have time, thanks

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py [hunk 1]
```diff
@@ -189,6 +190,21 @@ def end_time(self, val) -> None:
         """Set attribute end_time."""
         self._end_time = val
 
+    @property
+    def param_json(self) -> Optional[List[Dict]]:
+        """Return param json base on self.param."""
+        if self.param is None or not self.param:
```
Comment (devosend) [inline dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py]: Aren't these two judgment conditions the same?

Comment (zhongjiajie) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py]: It is two behavior, one is for `None` and other for `{}`(empty dict). But we can also combine to single statement `not self.param` too.

Comment (zhongjiajie) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py]: @devosend  WDYT, should we combine or separate

Comment (devosend) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py]: I think we should combine，beause and `None` and `{}` are empty. maybe we can add a comment.

Comment (zhongjiajie) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py]: I'm going to add this

Comment (zhongjiajie) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py]: I add f271f5b to correct it

Comment (devosend) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py]: LGTM

Comment (zhongjiajie) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py]: > LGTM

Thanks for your review

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**